lib/LTP/WhiteList.pm

```diff
@@ -49,44 +57,50 @@ sub download_whitelist {
 sub find_whitelist_entry {
     my ($env, $suite, $test) = @_;
 
-    my $path = get_var('LTP_KNOWN_ISSUES');
-    return undef unless defined($path);
-
-    my $content = path($path)->slurp;
-    my $issues  = Mojo::JSON::decode_json($content);
-    return undef unless $issues;
-    return undef unless exists $issues->{$suite};
+    $suite = find_whitelist_testsuite($suite);
+    return undef unless ($suite);
 
     my @issues;
-    if (ref($issues->{$suite}) eq 'ARRAY') {
-        @issues = @{$issues->{$suite}};
+    if (ref($suite) eq 'ARRAY') {
+        @issues = @{$suite};
     }
     else {
         $test =~ s/_postun$//g if check_var('KGRAFT', 1) && check_var('UNINSTALL_INCIDENT', 1);
-        return undef unless exists $issues->{$suite}->{$test};
-        @issues = @{$issues->{$suite}->{$test}};
+        return undef unless exists $suite->{$test};
+        @issues = @{$suite->{$test}};
     }
 
-  ISSUE:
     foreach my $cond (@issues) {
-        foreach my $filter (qw(product ltp_version revision arch kernel backend retval flavor)) {
-            next ISSUE if exists $cond->{$filter} and $env->{$filter} !~ m/$cond->{$filter}/;
-        }
-
-        return $cond;
+        return $cond if (whitelist_entry_match($cond, $env));
     }
 
     return undef;
 }
 
 sub override_known_failures {
     my ($self, $env, $suite, $test) = @_;
-    my $entry = find_whitelist_entry($env, $suite, $test);
+    my $entry;
+
+    if ($env->{retval} && ref($env->{retval}) eq 'ARRAY') {
+        my %local_env = %$env;
+
+        my @retvals = grep { $_ ne 0 } @{$env->{retval}};
+        return 0 unless (@retvals);
```

Comment:
Hmm, this is a bit more complicated. We have some whitelist entries with `retval: ^0$` for tests which pass but trigger a harmless kernel warning. You should still check for those if all retvals are zero.

Reply:
Lets just not exclude zero at all then!?

Reply:
That's the fun thing: This may not work either. If you have a retval array like `[0, 0, 0, 1, 0]` and there is a whitelist entry for `retval: ^1$` which is caused by a known random error, you *need* to ignore the zeroes.

Reply:
IOW: when `@retvals` is empty here, just search for whitelist entry with `retval => 0` instead of returning.

Reply:
ok

Reply:
plz check latest, thx